libs/dh/reports/feature-measurement-reports/src/components/request-report/request-report-modal.component.ts

```diff
@@ -89,16 +104,97 @@ export class DhRequestReportModal extends WattTypedModal<MeasurementsReportReque
   private readonly formBuilder = inject(NonNullableFormBuilder);
   private readonly environmentInjector = inject(EnvironmentInjector);
   private readonly destroyRef = inject(DestroyRef);
+  private readonly apollo = inject(Apollo);
+
+  private readonly toastService = inject(WattToastService);
+
+  private modal = viewChild.required(WattModalComponent);
 
   form: DhFormType = this.formBuilder.group({
-    period: new FormControl<WattRange<Date> | null>(null, [Validators.required]),
+    period: new FormControl<WattRange<Date> | null>(null, [
+      Validators.required,
+      startDateAndEndDateHaveSameMonthValidator(),
+    ]),
     gridAreas: new FormControl<string[] | null>(null, Validators.required),
   });
 
+  private gridAreaChanges = toSignal(this.form.controls.gridAreas.valueChanges);
+
   gridAreaOptions$ = this.getGridAreaOptions();
 
+  multipleGridAreasSelected = computed(() => {
+    const gridAreas = this.gridAreaChanges();
+
+    if (gridAreas == null) {
+      return false;
+    }
+
+    return gridAreas.length > 1;
+  });
+
   submitInProgress = signal(false);
 
+  // eslint-disable-next-line sonarjs/cognitive-complexity
+  submit(): void {
+    if (this.form.invalid || this.submitInProgress()) {
+      return;
+    }
+
+    const { period, gridAreas } = this.form.getRawValue();
+
+    if (period == null || gridAreas == null) {
+      return;
+    }
+
+    this.submitInProgress.set(true);
```

Comment:
Suggestion: Use state from the mutation help

Reply:
Done